### ISSUE TITLE
Unused shaping test fonts

### ISSUE THREAD

**Issue by RazrFalcon** [harfbuzz/harfbuzz#2098]
I've wrote a simple script to find unused fonts and actually found some.

```
data/aots/classdef1_font1.otf
data/aots/cmap0_font1.otf
data/aots/cmap10_font1.otf
data/aots/cmap10_font2.otf
data/aots/cmap12_font1.otf
data/aots/cmap14_font1.otf
data/aots/cmap2_font1.otf
data/aots/cmap4_font1.otf
data/aots/cmap4_font2.otf
data/aots/cmap4_font3.otf
data/aots/cmap4_font4.otf
data/aots/cmap6_font1.otf
data/aots/cmap6_font2.otf
data/aots/cmap8_font1.otf
data/aots/cmap_composition_font1.otf
data/aots/cmap_subtableselection_font1.otf
data/aots/cmap_subtableselection_font2.otf
data/aots/cmap_subtableselection_font3.otf
data/aots/cmap_subtableselection_font4.otf
data/aots/cmap_subtableselection_font5.otf
data/in-house/46669c8860cbfea13562a6ca0d83130ee571137b.ttf
data/text-rendering-tests/NotoSansKannada-Regular.ttf
data/text-rendering-tests/Selawik-README.md
data/text-rendering-tests/Selawik-variable.ttf
data/text-rendering-tests/TestCMAPMacTurkish.ttf
data/text-rendering-tests/TestGVAR-Composite-0.ttf
data/text-rendering-tests/TestGVAR-Composite-Missing.ttf
data/text-rendering-tests/TestShapeAran.ttf
data/text-rendering-tests/TestShapeKndaV3.ttf
data/text-rendering-tests/TestTRAKOne.ttf
```

The script itself:

```python
#!/usr/bin/env python

import os


IGNORE_TESTS = [
    'macos.tests',
]

def check(dir):
    test_files = sorted(os.listdir(dir + '/tests'))
    test_files = [f for f in test_files if f.endswith('.tests')]

    used_fonts = set()

    for file in test_files:
        if file in IGNORE_TESTS:
            continue

        with open(dir + '/tests/' + file, 'r') as f:
            for idx, test in enumerate(f.read().splitlines()):
                # check for fonts in comments too
                if test.startswith('#'):
                    test = test[1:]

                # skip empty lines
                if len(test) == 0:
                    continue

                fontfile = test.split(':')[0]
                fontfile = os.path.split(fontfile)[1]
                used_fonts.add(fontfile)

    font_files = set(os.listdir(dir + '/fonts'))
    for font in sorted(list(font_files.difference(used_fonts))):
        print(dir + '/' + font)


check('data/aots')
check('data/in-house')
check('data/text-rendering-tests')
```

It's not perfect, since it checks only tests and not all sources. So I've double-checked all the listed font via recursive grep.

Should we remove those files?

**Comment by ebraminio** [comment]
Can say at least aots ones are expected as we were supposed to see why https://github.com/adobe-type-tools/aots gsub/gpos tests results were different from ours, still having non-used but different fonts are still beneficial as they can be used as fuzzers we should however when they have added and why there aren't in use currently.

**Comment by khaledhosny** [comment]
The in-house test file fixed in #3082. The unused text-rendering-tests fonts are unused upstream, reported in https://github.com/unicode-org/text-rendering-tests/issues/70.

**Comment by khaledhosny** [comment]
For AOTS tests, `classdef1_font1.otf` is not used upstream. The cmap tests are not generated for HarfBuzz at all (upstream as well), so I’m not sure if we want test them. Either case, as @ebraminio said, even unused fonts help seed the fuzzer.
